extras/wait-for-bricks.sh

```diff
@@ -92,34 +48,36 @@ function volumes_waiting_for_bricks_up()
         local vol_down_info
         while read -r volname
         do
-                if [[ "Started" == "$(volinfo_field "$volname" 'statusStr')" ]]
+                read -r total_brick_count
+                brick_count="$(volume_online_brick_count "$volname" "$total_brick_count")"
+                if [[ $brick_count -ne $total_brick_count ]]
                 then
-                        if [[ "Y" == "$(is_enabled "$(volinfo_option "$volname" features.shard)")" ]]
+                        vol_down_info="$volname ($((total_brick_count - brick_count))/$total_brick_count)"
+                        if [[ -z "$wait_info" ]]
                         then
-                                brick_count="$(volume_online_brick_count "$volname")"
-                                total_brick_count=$(volinfo_field "$volname" 'brickCount')
-                                if [[ $brick_count -ne $total_brick_count ]]
-                                then
-                                        vol_down_info="$volname ($((total_brick_count - brick_count))/$total_brick_count)"
-                                        if [[ -z "$wait_info" ]]
-                                        then
-                                                wait_info="$vol_down_info"
-                                        else
-                                                wait_info="$wait_info, $vol_down_info"
-                                        fi
-                                fi
+                                wait_info="$vol_down_info"
+                        else
+                                wait_info="$wait_info, $vol_down_info"
                         fi
                 fi
-        done < <(gluster volume list)
+        done < <(block_volnames_brick_count_get)
         if [[ ! -z "$wait_info" ]]; then echo "$wait_info"; fi
 }
 
 function check_glusterd_running()
 {
-        if ! pidof glusterd > /dev/null 2>&1
+        if ! command -v ps > /dev/null 2>&1
+        then
+                printLog "ERROR: 'ps' command is not found"
+                exit 1
+        fi
+
+# Explanation of why ps command is used this way:
+# https://stackoverflow.com/questions/9117507/linux-unix-command-to-determine-if-process-is-running
```

Comment:
Not exactly. This is a way tweaked version of what is been explained at stack-overflow, better remove it IMO.

Reply:
In one of the comments on that page:

> grep will also find itself running (e.g. ps cax | grep randomname will always return 0 because grep finds grep randomname (hope this is clear...). One fix is to add square brackets around the first letter of the process name, e.g. ps cax | grep [r]andomname. – Kyle G. Oct 29 '13 at 3:19

I think the only extra flag we added is -w for grep. I would like to keep it so that people who read this code would know why it is done this way.

Reply:
I did not refer that part then. I refered our replace-node script for [] usage. Cool. I'm all okay